OptimizelySDK/Odp/IOdpManager.cs

```diff
@@ -0,0 +1,65 @@
+﻿/* 
+ * Copyright 2022 Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System.Collections.Generic;
+
+namespace OptimizelySDK.Odp
+{
+    /// <summary>
+    /// Interface describing orchestration of segment manager, event manager, and ODP config
+    /// </summary>
+    public interface IOdpManager
+    {
+        /// <summary>
+        /// Update the settings being used for ODP configuration and reset/restart dependent processes
+        /// </summary>
+        /// <param name="apiKey">Public API key from ODP</param>
+        /// <param name="apiHost">Host portion of the URL of ODP</param>
+        /// <param name="segmentsToCheck">Audience segments to consider</param>
+        /// <returns>True if settings were update otherwise False</returns>
+        bool UpdateSettings(string apiKey, string apiHost, List<string> segmentsToCheck);
+
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="options">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        List<string> FetchQualifiedSegments(string userId, List<OdpSegmentOption> options);
```

Comment:
nit: This should be `Array`

Reply:
Oh, yes. Sorry.  Old habit from previous work.